app/src/pages/authoring/[resourceType]/[id].tsx

```diff
@@ -31,11 +77,26 @@ export default function ResourceAuthoringPage() {
         identifier => identifier.system === 'http://hl7.org/fhir/cqi/ecqm/Measure/Identifier/cms'
       );
       if (cmsIdentifier?.value) {
-        setIdentifier(cmsIdentifier.value);
+        setIdentifierValue(cmsIdentifier.value);
+        if (cmsIdentifier.system) {
+          setIdentifierSystem(cmsIdentifier.system);
+        }
       } else if (resource.identifier[0].value) {
-        setIdentifier(resource.identifier[0].value);
+        setIdentifierValue(resource.identifier[0].value);
+        if (resource.identifier[0].system) {
+          setIdentifierSystem(resource.identifier[0].value);
+        }
```

Comment:
I've run into an issue where if you go to edit a resource without a value or system identifier and add both of them to the resource, the `system identifier` gets overwritten with the `value identifier`. Also, if you delete the value identifier but don't delete the system identifier, the identifier will be deleted from the json entirely.

Reply:
Nice catch with the overwritten stuff!! For the other part of the comment, I believe that is actually the desired functionality since we just don't want the user to be able to define only an identifier system- the user should only be able to enter one of three things:
- nothing for either identifier.system or identifier.value
- just identifier.value
- both identifier.system and identifier.value